### ISSUE TITLE
Elektrad + Go bindings CI / Build Integration

### ISSUE THREAD

**Issue by raphi011** [ElektraInitiative/libelektra#2962]
Add Elektrad to the CMake build. Unfortunately I'm having huge problems accomplishing this, see [this](https://github.com/ElektraInitiative/libelektra/pull/2827#discussion_r322008374) discussion.

Add go-bindings to the test pipeline (e.g. as git submodule)

Run tests on Travis-CI after each push .. blocked by https://github.com/ElektraInitiative/libelektra/pull/2932

**Comment by markus2330** [comment]
Just to make sure it is not overseen (I have already written it in email conversation several times but @sanssecours might not know about it) https://github.com/cpconduce/go_cmake might make the go integration task much easier.

**Comment by raphi011** [comment]
@markus2330 @sanssecours , I'm trying to create a temporary elektra.pc file but not entirely there. I still get linker issues and I'm not sure why. 

I'm using [this dockerfile](https://gist.github.com/raphi011/a605e8eac19e040558932c957e745d95)  and [this commit](https://github.com/raphi011/libelektra/commit/f2261021a5bb11cc6355db1a88bc8fda244c07d1). Any ideas?

Error message

```
# github.com/ElektraInitiative/go-elektra/kdb
/tmp/go-build549156878/b039/_x002.o: In function `_cgo_ea9ba2b9169f_Cfunc_kdbClose':
/tmp/go-build/cgo-gcc-prolog:54: undefined reference to `kdbClose'
/tmp/go-build549156878/b039/_x002.o: In function `_cgo_ea9ba2b9169f_Cfunc_kdbGet':
/tmp/go-build/cgo-gcc-prolog:75: undefined reference to `kdbGet'
/tmp/go-build549156878/b039/_x002.o: In function `_cgo_ea9ba2b9169f_Cfunc_kdbOpen':
/tmp/go-build/cgo-gcc-prolog:93: undefined reference to `kdbOpen'
/tmp/go-build549156878/b039/_x002.o: In function `_cgo_ea9ba2b9169f_Cfunc_kdbSet':
/tmp/go-build/cgo-gcc-prolog:114: undefined reference to `kdbSet'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyBaseName':
/tmp/go-build/cgo-gcc-prolog:64: undefined reference to `keyBaseName'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyDel':
/tmp/go-build/cgo-gcc-prolog:83: undefined reference to `keyDel'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyDup':
/tmp/go-build/cgo-gcc-prolog:101: undefined reference to `keyDup'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyGetBinary':
/tmp/go-build/cgo-gcc-prolog:121: undefined reference to `keyGetBinary'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyGetMeta':
/tmp/go-build/cgo-gcc-prolog:140: undefined reference to `keyGetMeta'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyGetValueSize':
/tmp/go-build/cgo-gcc-prolog:158: undefined reference to `keyGetValueSize'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyIsBelowOrSame':
/tmp/go-build/cgo-gcc-prolog:178: undefined reference to `keyIsBelowOrSame'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyIsDirectBelow':
/tmp/go-build/cgo-gcc-prolog:198: undefined reference to `keyIsDirectBelow'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyName':
/tmp/go-build/cgo-gcc-prolog:216: undefined reference to `keyName'
/tmp/go-build549156878/b039/_x003.o: In function `keyNewEmptyWrapper':
/root/go/pkg/mod/github.com/!elektra!initiative/go-elektra@v0.0.0-20190904191204-e2bdbf4a0377/kdb/key.go:7: undefined reference to `keyNew'
/tmp/go-build549156878/b039/_x003.o: In function `keyNewValueWrapper':
/root/go/pkg/mod/github.com/!elektra!initiative/go-elektra@v0.0.0-20190904191204-e2bdbf4a0377/kdb/key.go:15: undefined reference to `keyNew'
/tmp/go-build549156878/b039/_x003.o: In function `keyNewWrapper':
/root/go/pkg/mod/github.com/!elektra!initiative/go-elektra@v0.0.0-20190904191204-e2bdbf4a0377/kdb/key.go:11: undefined reference to `keyNew'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyNextMeta':
/tmp/go-build/cgo-gcc-prolog:288: undefined reference to `keyNextMeta'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyRewindMeta':
/tmp/go-build/cgo-gcc-prolog:307: undefined reference to `keyRewindMeta'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keySetBinary':
/tmp/go-build/cgo-gcc-prolog:327: undefined reference to `keySetBinary'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keySetMeta':
/tmp/go-build/cgo-gcc-prolog:347: undefined reference to `keySetMeta'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keySetName':
/tmp/go-build/cgo-gcc-prolog:366: undefined reference to `keySetName'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keySetString':
/tmp/go-build/cgo-gcc-prolog:385: undefined reference to `keySetString'
/tmp/go-build549156878/b039/_x003.o: In function `_cgo_ea9ba2b9169f_Cfunc_keyString':
/tmp/go-build/cgo-gcc-prolog:403: undefined reference to `keyString'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksAppend':
/tmp/go-build/cgo-gcc-prolog:53: undefined reference to `ksAppend'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksAppendKey':
/tmp/go-build/cgo-gcc-prolog:72: undefined reference to `ksAppendKey'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksClear':
/tmp/go-build/cgo-gcc-prolog:91: undefined reference to `ksClear'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksCopy':
/tmp/go-build/cgo-gcc-prolog:111: undefined reference to `ksCopy'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksCut':
/tmp/go-build/cgo-gcc-prolog:130: undefined reference to `ksCut'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksDel':
/tmp/go-build/cgo-gcc-prolog:149: undefined reference to `ksDel'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksGetSize':
/tmp/go-build/cgo-gcc-prolog:167: undefined reference to `ksGetSize'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksHead':
/tmp/go-build/cgo-gcc-prolog:185: undefined reference to `ksHead'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksLookup':
/tmp/go-build/cgo-gcc-prolog:206: undefined reference to `ksLookup'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksLookupByName':
/tmp/go-build/cgo-gcc-prolog:227: undefined reference to `ksLookupByName'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksNeedSync':
/tmp/go-build/cgo-gcc-prolog:246: undefined reference to `ksNeedSync'
/tmp/go-build549156878/b039/_x004.o: In function `ksNewWrapper':
/root/go/pkg/mod/github.com/!elektra!initiative/go-elektra@v0.0.0-20190904191204-e2bdbf4a0377/kdb/keyset.go:7: undefined reference to `ksNew'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksNext':
/tmp/go-build/cgo-gcc-prolog:282: undefined reference to `ksNext'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksPop':
/tmp/go-build/cgo-gcc-prolog:300: undefined reference to `ksPop'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksRewind':
/tmp/go-build/cgo-gcc-prolog:319: undefined reference to `ksRewind'
/tmp/go-build549156878/b039/_x004.o: In function `_cgo_ea9ba2b9169f_Cfunc_ksTail':
/tmp/go-build/cgo-gcc-prolog:337: undefined reference to `ksTail'
collect2: error: ld returned 1 exit status
# github.com/ElektraInitiative/go-elektra/kdb
/root/go/pkg/mod/github.com/!elektra!initiative/go-elektra@v0.0.0-20190904191204-e2bdbf4a0377/kdb/keyset.go: In function 'ksNewWrapper':
/root/go/pkg/mod/github.com/!elektra!initiative/go-elektra@v0.0.0-20190904191204-e2bdbf4a0377/kdb/keyset.go:7:4: warning: missing sentinel in function call [-Wformat=]
 //   return ksNew(size, KEY_END);
    ^
src/tools/web/CMakeFiles/elektrad.dir/build.make:65: recipe for target 'src/tools/web/elektrad/elektrad' failed
make[2]: *** [src/tools/web/elektrad/elektrad] Error 2
CMakeFiles/Makefile2:13596: recipe for target 'src/tools/web/CMakeFiles/elektrad.dir/all' failed
make[1]: *** [src/tools/web/CMakeFiles/elektrad.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by raphi011** [comment]
@markus2330 ping :)

**Comment by markus2330** [comment]
`-L${libdir}` is there but `-lelektra-core ...` is missing (see other pkg-config files)

**Comment by raphi011** [comment]
> `-L${libdir}` is there but `-lelektra-core ...` is missing (see other pkg-config files)

sometimes life could be so easy... thanks

**Comment by raphi011** [comment]
The Mac CLang build now runs the web tests

**Comment by markus2330** [comment]
Can you open a new issue for the CI integration of Linux (once we have the Ubuntu packages)?
